### DISCUSSION TITLE
New features fo the node palette.

### DISCUSSION THREAD

**Discussion by marco-dmaa** [FredBlo/node-red-contrib-myhome-bticino-v2#18] [Ideas]
Hi,
i want to thank you for your work, the nodes works very well and the connection to the gateways is so stable.

Did you ever think of add the feature to change the gateway parameters (IP Address and Open Password) dynamically from a message?
Es: msg.gateway = xxx.xxx.xxx.xxx and msg.password = xxxxx

This would be a very interesting mod...

Thanks, Marco.

**Comment by FredBlo** [comment]
Hi @marco-dmaa ,

I never heard such need, what is your use case which makes it interesting ?

Anyway, technically, such a change would require to refactor the whole structure.
I am even not sure this would be feasible because all nodes are linked to the gateway. Changing the IP (and pwd) would actually require to unload all listening nodes and restart them, which is actually what Node-RED does when a (re)deploy occurs...
